src/main/java/core/basesyntax/service/operation/SupplyOperationHandler.java

```diff
@@ -0,0 +1,14 @@
+package core.basesyntax.service.operation;
+
+import core.basesyntax.service.FruitTransaction;
+import java.util.Map;
+
+public class SupplyOperationHandler implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction, Map<String, Integer> storage) {
```

Comment:
According to SOLID principles, your service classes should be independent. You should not pass the storage map to the `handle` method. Instead, you should create a separate `StorageService` that will be responsible for all operations with storage. It will improve your code structure and make it more flexible in case of any changes in the future.